{"context": "\n\n\u4ee5\u4e0b\u306e\u8a18\u8ff0\u3092\u767a\u898b\n\n\npandas 0.18.0 documentation \u00bb Remote Data Access\n\n\nWarning In pandas 0.17.0, the sub-package pandas.io.data will be removed in favor of a separately installable pandas-datareader package. This will allow the data modules to be independently updated to your pandas installation. The API for pandas-datareader v0.1.1 is the same as in pandas v0.16.1. (GH8961)\nYou should replace the imports of the following:\nfrom pandas.io import data, wb\n\nWith:\nfrom pandas_datareader import data, wb\n\n\n\nAnaconda 3.5.1\u74b0\u5883 \u3067 pandas.io \u3092\u30a4\u30f3\u30dd\u30fc\u30c8\u3057\u3088\u3046\u3068\u3059\u308b\u3068\u3001\u8fd1\u3005\u524a\u9664\uff08remove\uff09\u4e88\u5b9a\u3001\u3068\u8868\u793a\u3055\u308c\u308b\n\n\nMacBook Terminal\n~ $\n~ $ python --version\nPython 3.5.1 :: Anaconda 2.4.1 (x86_64)\n~ $\n\n\n\nPython3\n>>>\n>>> from pandas.io import data\n/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas/io/data.py:33: FutureWarning:\nThe pandas.io.data module is moved to a separate package (pandas-datareader) and will be removed from pandas in a future version.\nAfter installing the pandas-datareader package (https://github.com/pydata/pandas-datareader), you can change the import ``from pandas.io import data, wb`` to ``from pandas_datareader import data, wb``.\n  FutureWarning)\n>>>\n\n\n\n\nAnaconda \u74b0\u5883 \u306b pandas_datareader\u30e2\u30b8\u30e5\u30fc\u30eb \u3092 \u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\n\n\n\uff08\u53c2\u8003\uff09Anaconda \u3078\u306e \u30e2\u30b8\u30e5\u30fc\u30eb\u8ffd\u52a0\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u65b9\u6cd5\n\n\n\nPython\u6570\u5024\u8a08\u7b97\u74b0\u5883Anaconda\u306e\u5c0e\u5165 \u300c3. Anaconda\u306b\u30d1\u30c3\u30b1\u30fc\u30b8\u3092\u8ffd\u52a0\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u300d\n\n\nAnaconda\u3078\u8ffd\u52a0\u3067\u30d1\u30c3\u30b1\u30fc\u30b8\u3092\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3059\u308b\u65b9\u6cd5\u306f\uff08\uff11\uff09\u300cconda install\u300d\u30b3\u30de\u30f3\u30c9\u3092\u4f7f\u3046\u304b\u3001\uff082\uff09\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3057\u305f\u3044\u30d1\u30c3\u30b1\u30fc\u30b8\u306e\u30d5\u30a1\u30a4\u30eb\u7fa4\u306b\u540c\u68b1\u3055\u308c\u3066\u3044\u308b\u300csetup.py\u300d\u3092\u4f7f\u3063\u3066\u300cpython setup.py install\u300d\u30b3\u30de\u30f3\u30c9\u3092\u5b9f\u884c\u3059\u308b\u304b\u3001\uff083\uff09pip\u30b3\u30de\u30f3\u30c9\u3092\u4f7f\u3046\u65b9\u6cd5\u306e3\u901a\u308a\u3042\u308a\u307e\u3059\u3002\n\n\n\npandas_datareader \u306f\u898b\u3064\u304b\u3089\u306a\u3044\u304c\u3001pandas-datareader \u306a\u3089\u3042\u308b\u3001\u3068\u6307\u6458\u3055\u308c\u308b\n\n\nMacBook Terminal\n~ $ conda install pandas_datareader\nFetching package metadata: ....\nSolving package specifications: .\nError:  Package missing in current osx-64 channels:\n  - pandas_datareader\n\nDid you mean one of these?\n\n    pandas-datareader\n\nYou can search for this package on anaconda.org with\n\n    anaconda search -t conda pandas_datareader\n~ $\n\n\n\npandas-datareader \u3092 conda install\n\n\nMacBook Terminal\n~ $ conda install pandas-datareader\nFetching package metadata: ....\nSolving package specifications: .........\n\nPackage plan for installation in environment /Users/hirofumi.yashima/anaconda:\n\nThe following packages will be downloaded:\n\n    package                    |            build\n    ---------------------------|-----------------\n    pandas-datareader-0.2.0    |           py35_0          40 KB\n\nThe following NEW packages will be INSTALLED:\n\n    pandas-datareader: 0.2.0-py35_0\n\nProceed ([y]/n)? y\n\nFetching packages ...\npandas-datarea 100% |#####################################################################################| Time: 0:00:00 118.56 kB/s\nExtracting packages ...\n[      COMPLETE      ]|########################################################################################################| 100%\nLinking packages ...\n[      COMPLETE      ]|########################################################################################################| 100%\n~ $\n\n\n\nMacBook Terminal\n~ $ conda list | grep pandas-datareader\npandas-datareader         0.2.0                    py35_0\n~ $\n~ $ conda list | grep pandas_datareader\n~ $\n\n\n\nimport\u5b9f\u884c\u6642\u306f\u3001pandas-datareader\uff08\u30cf\u30a4\u30d5\u30f3\uff09 \u3067\u306f\u306a\u304f\u3001pandas_datareader\uff08\u30a2\u30f3\u30c0\u30fc\u30b9\u30b3\u30a2\uff09 \u306b\u306a\u308b\n\n\nPython3\n>>> import pandas_datareader.data as web\n>>> import pandas-datareader-0.2.0.data as web\n  File \"<stdin>\", line 1\n    import pandas-datareader-0.2.0.data as web\n                 ^\nSyntaxError: invalid syntax\n>>>\n>>> import pandas-datareader\n  File \"<stdin>\", line 1\n    import pandas-datareader\n                 ^\nSyntaxError: invalid syntax\n>>>\n\n\n\nimport pandas_datareader.data as web \u3067\u30a4\u30f3\u30dd\u30fc\u30c8\u3057\u3066\u3001yahoo\u682a\u4fa1\u3092\u53d6\u5f97\u3057\u3066\u307f\u308b\n\n\nPython3\n>>>\n>>> import pandas_datareader.data as web\n>>> import datetime as dt\n>>>\n>>> start = dt.datetime(2010,1,1)\n>>> end = dt.datetime(2015,4,1)\n>>>\n>>> f = web.DataReader(\"F\", 'yahoo', start, end)\n>>>\n>>> import types\n>>> print(type(f))\n<class 'pandas.core.frame.DataFrame'>\n>>>\n>>> print(f.columns)\nIndex(['Open', 'High', 'Low', 'Close', 'Volume', 'Adj Close'], dtype='object')\n>>>\n>>> print(f.ix['2010-01-04'])\nOpen               10.170000\nHigh               10.280000\nLow                10.050000\nClose              10.280000\nVolume       60855800.000000\nAdj Close           8.852526\nName: 2010-01-04 00:00:00, dtype: float64\n>>>\n\n\n\nTuyorial\u30b3\u30fc\u30c9\u306b\u306f\u306a\u3044\u304c\u3001FTSE\u306e\u682a\u4fa1\u30c7\u30fc\u30bf\u3082\u53d6\u5f97\u3057\u3066\u307f\u308b\n\n>>>\n>>> FTSE_stock_prices = web.get_data_yahoo('^FTSE', start=start, end=end)\n>>> print(FTSE_stock_prices.shape)\n(1361, 6)\n>>>\n>>> print(FTSE_stock_prices.ix[0:6, :])\n                   Open         High          Low        Close      Volume  \\\nDate\n2010-01-01  5412.899902  5412.899902  5412.899902  5412.899902           0\n2010-01-04  5412.899902  5500.299805  5410.799805  5500.299805   750942000\n2010-01-05  5500.299805  5536.399902  5480.700195  5522.500000  1149301200\n2010-01-06  5522.500000  5536.500000  5497.700195  5530.000000   998295300\n2010-01-07  5530.000000  5551.700195  5499.799805  5526.700195  1162933700\n2010-01-08  5526.700195  5549.299805  5494.799805  5534.200195  1006420600\n\n              Adj Close\nDate\n2010-01-01  5412.899902\n2010-01-04  5500.299805\n2010-01-05  5522.500000\n2010-01-06  5530.000000\n2010-01-07  5526.700195\n2010-01-08  5534.200195\n>>>\n\n\n\u7d9a\u3051\u3066\u3001\u89e3\u8aac\u30a6\u30a7\u30d6\u30da\u30fc\u30b8\u306e\u30c1\u30e5\u30fc\u30c8\u30ea\u30a2\u30eb\u30fb\u30b3\u30fc\u30c9\u3092\u6d41\u3057\u3066\u307f\u308b\n\n\nOptions\u30c7\u30fc\u30bf \u306e \u53d6\u308a\u8fbc\u307f\n\n\nPython3\n>>>\n>>> import pandas_datareader.data.Options\nTraceback (most recent call last):\n  File \"<stdin>\", line 1, in <module>\nImportError: No module named 'pandas_datareader.data.Options'; 'pandas_datareader.data' is not a package\n>>>\n>>> from pandas_datareader.data import Options\n>>>\n>>> aapl = Options('aapl', 'yahoo')\n>>> print(type(aapl))\n<class 'pandas_datareader.yahoo.options.Options'>\n>>>\n\n\n\nget_all_data\u30e1\u30bd\u30c3\u30c9\u3092\u547c\u3073\u51fa\u3057\n\n\nPython3\n>>> data = aapl.get_all_data()\nTraceback (most recent call last):\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas_datareader/yahoo/options.py\", line 593, in _get_data_in_date_range\n    frame = getattr(self, nam)\nAttributeError: 'Options' object has no attribute 'calls160408'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas_datareader/yahoo/options.py\", line 216, in _get_option_data\n    frames = getattr(self, frame_name)\nAttributeError: 'Options' object has no attribute '_frames160408'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"<stdin>\", line 1, in <module>\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas_datareader/yahoo/options.py\", line 581, in get_all_data\n    return self._get_data_in_date_range(dates=expiry_dates, call=call, put=put)\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas_datareader/yahoo/options.py\", line 595, in _get_data_in_date_range\n    frame = self._get_option_data(expiry=expiry_date, name=name)\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas_datareader/yahoo/options.py\", line 218, in _get_option_data\n    frames = self._get_option_frames_from_yahoo(expiry)\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas_datareader/yahoo/options.py\", line 138, in _get_option_frames_from_yahoo\n    option_frames = self._option_frames_from_url(url)\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas_datareader/yahoo/options.py\", line 159, in _option_frames_from_url\n    frames = read_html(url)\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas/io/html.py\", line 866, in read_html\n    parse_dates, tupleize_cols, thousands, attrs, encoding)\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas/io/html.py\", line 718, in _parse\n    parser = _parser_dispatch(flav)\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas/io/html.py\", line 663, in _parser_dispatch\n    raise ImportError(\"html5lib not found, please install it\")\nImportError: html5lib not found, please install it\n>>>\n\n\n\nget_all_data\u30e1\u30bd\u30c3\u30c9\u3092\u547c\u3073\u51fa\u3059\u3068\u3001html5lib\u30e2\u30b8\u30e5\u30fc\u30eb\u304c\u306a\u3044\u3068\u8a00\u308f\u308c\u308b\u3002\n\n\nPython3\n>>> import html5lib\nTraceback (most recent call last):\n  File \"<stdin>\", line 1, in <module>\nImportError: No module named 'html5lib'\n>>>\n>>> quit()\n\n\n\nconda install \u3067 Anaconda \u306b \u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3059\u308b\n\n\nMacBook Terminal\n~ $\n~ $ conda install html5lib\nFetching package metadata: ....\nSolving package specifications: .........\n\nPackage plan for installation in environment\n\n\n\n\u4eca\u5ea6\u306f\u3001get_all_data\u30e1\u30bd\u30c3\u30c9 \u3092 \u5b9f\u884c\u3067\u304d\u305f\n\n\nPython3\n>>> import html5lib\n>>> from pandas_datareader.data import Options\n>>>\n>>> aapl = Options('aapl', 'yahoo')\n>>> data = aapl.get_all_data()\n>>> print(type(data))\n<class 'pandas.core.frame.DataFrame'>\n>>>\n>>> print(data.columns)\nIndex(['Last', 'Bid', 'Ask', 'Chg', 'PctChg', 'Vol', 'Open_Int', 'IV', 'Root',\n       'IsNonstandard', 'Underlying', 'Underlying_Price', 'Quote_Time'],\n      dtype='object')\n>>>\n>>> print(data.iloc[0:5, 0:5])\n                                            Last  Bid    Ask  Chg PctChg\nStrike Expiry     Type Symbol\n10.0   2016-06-17 call AAPL160617C00010000     0   91  91.45    0  0.00%\n                  put  AAPL160617P00010000     0    0   0.03    0  0.00%\n12.5   2016-06-17 put  AAPL160617P00012500     0    0   0.03    0  0.00%\n15.0   2016-06-17 call AAPL160617C00015000     0   86  86.45    0  0.00%\n                  put  AAPL160617P00015000     0    0   0.03    0  0.00%\n>>>\n\n\n##__\u4ee5\u4e0b\u306e\u8a18\u8ff0\u3092\u767a\u898b__\n\n* [pandas 0.18.0 documentation \u00bb Remote Data Access](http://pandas.pydata.org/pandas-docs/stable/remote_data.html)\n\n> Warning In pandas 0.17.0, the sub-package pandas.io.data will be removed in favor of a separately installable pandas-datareader package. This will allow the data modules to be independently updated to your pandas installation. The API for pandas-datareader v0.1.1 is the same as in pandas v0.16.1. (GH8961)\n>You should replace the imports of the following:\n>\n>```\n>from pandas.io import data, wb\n>```\n>\n>With:\n>\n>```\n>from pandas_datareader import data, wb\n>```\n\n###__Anaconda 3.5.1\u74b0\u5883 \u3067 pandas.io \u3092\u30a4\u30f3\u30dd\u30fc\u30c8\u3057\u3088\u3046\u3068\u3059\u308b\u3068\u3001\u8fd1\u3005\u524a\u9664\uff08remove\uff09\u4e88\u5b9a\u3001\u3068\u8868\u793a\u3055\u308c\u308b__\n\n```{bash:MacBook Terminal}\n~ $\n~ $ python --version\nPython 3.5.1 :: Anaconda 2.4.1 (x86_64)\n~ $\n```\n\n\n```{ython:Python3}\n>>>\n>>> from pandas.io import data\n/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas/io/data.py:33: FutureWarning:\nThe pandas.io.data module is moved to a separate package (pandas-datareader) and will be removed from pandas in a future version.\nAfter installing the pandas-datareader package (https://github.com/pydata/pandas-datareader), you can change the import ``from pandas.io import data, wb`` to ``from pandas_datareader import data, wb``.\n  FutureWarning)\n>>>\n```\n\n___\n\n###__Anaconda \u74b0\u5883 \u306b pandas_datareader\u30e2\u30b8\u30e5\u30fc\u30eb \u3092 \u30a4\u30f3\u30b9\u30c8\u30fc\u30eb__\n\n####__\uff08\u53c2\u8003\uff09Anaconda \u3078\u306e \u30e2\u30b8\u30e5\u30fc\u30eb\u8ffd\u52a0\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u65b9\u6cd5__\n\n___\n\n* [Python\u6570\u5024\u8a08\u7b97\u74b0\u5883Anaconda\u306e\u5c0e\u5165 \u300c3. Anaconda\u306b\u30d1\u30c3\u30b1\u30fc\u30b8\u3092\u8ffd\u52a0\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u300d](http://morimori2008.web.fc2.com/contents/PCprograming/python/pythonAnaconda.html)\n\n> Anaconda\u3078\u8ffd\u52a0\u3067\u30d1\u30c3\u30b1\u30fc\u30b8\u3092\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3059\u308b\u65b9\u6cd5\u306f\uff08\uff11\uff09\u300cconda install\u300d\u30b3\u30de\u30f3\u30c9\u3092\u4f7f\u3046\u304b\u3001\uff082\uff09\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3057\u305f\u3044\u30d1\u30c3\u30b1\u30fc\u30b8\u306e\u30d5\u30a1\u30a4\u30eb\u7fa4\u306b\u540c\u68b1\u3055\u308c\u3066\u3044\u308b\u300csetup.py\u300d\u3092\u4f7f\u3063\u3066\u300cpython setup.py install\u300d\u30b3\u30de\u30f3\u30c9\u3092\u5b9f\u884c\u3059\u308b\u304b\u3001\uff083\uff09pip\u30b3\u30de\u30f3\u30c9\u3092\u4f7f\u3046\u65b9\u6cd5\u306e3\u901a\u308a\u3042\u308a\u307e\u3059\u3002\n\n___\n\n* __pandas_datareader \u306f\u898b\u3064\u304b\u3089\u306a\u3044\u304c\u3001pandas-datareader \u306a\u3089\u3042\u308b\u3001\u3068\u6307\u6458\u3055\u308c\u308b__\n\n\n```{bash:MacBook Terminal}\n~ $ conda install pandas_datareader\nFetching package metadata: ....\nSolving package specifications: .\nError:  Package missing in current osx-64 channels:\n  - pandas_datareader\n\nDid you mean one of these?\n\n    pandas-datareader\n\nYou can search for this package on anaconda.org with\n\n    anaconda search -t conda pandas_datareader\n~ $\n```\n\n\n* __pandas-datareader \u3092 conda install__\n\n\n```{bash:MacBook Terminal}\n~ $ conda install pandas-datareader\nFetching package metadata: ....\nSolving package specifications: .........\n\nPackage plan for installation in environment /Users/hirofumi.yashima/anaconda:\n\nThe following packages will be downloaded:\n\n    package                    |            build\n    ---------------------------|-----------------\n    pandas-datareader-0.2.0    |           py35_0          40 KB\n\nThe following NEW packages will be INSTALLED:\n\n    pandas-datareader: 0.2.0-py35_0\n\nProceed ([y]/n)? y\n\nFetching packages ...\npandas-datarea 100% |#####################################################################################| Time: 0:00:00 118.56 kB/s\nExtracting packages ...\n[      COMPLETE      ]|########################################################################################################| 100%\nLinking packages ...\n[      COMPLETE      ]|########################################################################################################| 100%\n~ $\n```\n\n```{bash:MacBook Terminal}\n~ $ conda list | grep pandas-datareader\npandas-datareader         0.2.0                    py35_0\n~ $\n~ $ conda list | grep pandas_datareader\n~ $\n```\n\n\n* __import\u5b9f\u884c\u6642\u306f\u3001pandas-datareader\uff08\u30cf\u30a4\u30d5\u30f3\uff09 \u3067\u306f\u306a\u304f\u3001pandas_datareader\uff08\u30a2\u30f3\u30c0\u30fc\u30b9\u30b3\u30a2\uff09 \u306b\u306a\u308b__\n\n```{python:Python3}\n>>> import pandas_datareader.data as web\n>>> import pandas-datareader-0.2.0.data as web\n  File \"<stdin>\", line 1\n    import pandas-datareader-0.2.0.data as web\n                 ^\nSyntaxError: invalid syntax\n>>>\n>>> import pandas-datareader\n  File \"<stdin>\", line 1\n    import pandas-datareader\n                 ^\nSyntaxError: invalid syntax\n>>>\n```\n\n* __import pandas_datareader.data as web \u3067\u30a4\u30f3\u30dd\u30fc\u30c8\u3057\u3066\u3001yahoo\u682a\u4fa1\u3092\u53d6\u5f97\u3057\u3066\u307f\u308b__\n\n```{python:Python3}\n>>>\n>>> import pandas_datareader.data as web\n>>> import datetime as dt\n>>>\n>>> start = dt.datetime(2010,1,1)\n>>> end = dt.datetime(2015,4,1)\n>>>\n>>> f = web.DataReader(\"F\", 'yahoo', start, end)\n>>>\n>>> import types\n>>> print(type(f))\n<class 'pandas.core.frame.DataFrame'>\n>>>\n>>> print(f.columns)\nIndex(['Open', 'High', 'Low', 'Close', 'Volume', 'Adj Close'], dtype='object')\n>>>\n>>> print(f.ix['2010-01-04'])\nOpen               10.170000\nHigh               10.280000\nLow                10.050000\nClose              10.280000\nVolume       60855800.000000\nAdj Close           8.852526\nName: 2010-01-04 00:00:00, dtype: float64\n>>>\n```\n\n* __Tuyorial\u30b3\u30fc\u30c9\u306b\u306f\u306a\u3044\u304c\u3001FTSE\u306e\u682a\u4fa1\u30c7\u30fc\u30bf\u3082\u53d6\u5f97\u3057\u3066\u307f\u308b__\n\n```\n>>>\n>>> FTSE_stock_prices = web.get_data_yahoo('^FTSE', start=start, end=end)\n>>> print(FTSE_stock_prices.shape)\n(1361, 6)\n>>>\n>>> print(FTSE_stock_prices.ix[0:6, :])\n                   Open         High          Low        Close      Volume  \\\nDate\n2010-01-01  5412.899902  5412.899902  5412.899902  5412.899902           0\n2010-01-04  5412.899902  5500.299805  5410.799805  5500.299805   750942000\n2010-01-05  5500.299805  5536.399902  5480.700195  5522.500000  1149301200\n2010-01-06  5522.500000  5536.500000  5497.700195  5530.000000   998295300\n2010-01-07  5530.000000  5551.700195  5499.799805  5526.700195  1162933700\n2010-01-08  5526.700195  5549.299805  5494.799805  5534.200195  1006420600\n\n              Adj Close\nDate\n2010-01-01  5412.899902\n2010-01-04  5500.299805\n2010-01-05  5522.500000\n2010-01-06  5530.000000\n2010-01-07  5526.700195\n2010-01-08  5534.200195\n>>>\n```\n\n\n#### __\u7d9a\u3051\u3066\u3001\u89e3\u8aac\u30a6\u30a7\u30d6\u30da\u30fc\u30b8\u306e\u30c1\u30e5\u30fc\u30c8\u30ea\u30a2\u30eb\u30fb\u30b3\u30fc\u30c9\u3092\u6d41\u3057\u3066\u307f\u308b__\n\n* __Options\u30c7\u30fc\u30bf \u306e \u53d6\u308a\u8fbc\u307f__\n\n```{python:Python3}\n>>>\n>>> import pandas_datareader.data.Options\nTraceback (most recent call last):\n  File \"<stdin>\", line 1, in <module>\nImportError: No module named 'pandas_datareader.data.Options'; 'pandas_datareader.data' is not a package\n>>>\n>>> from pandas_datareader.data import Options\n>>>\n>>> aapl = Options('aapl', 'yahoo')\n>>> print(type(aapl))\n<class 'pandas_datareader.yahoo.options.Options'>\n>>>\n```\n\n* __get_all_data\u30e1\u30bd\u30c3\u30c9\u3092\u547c\u3073\u51fa\u3057__\n\n```{python:Python3}\n>>> data = aapl.get_all_data()\nTraceback (most recent call last):\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas_datareader/yahoo/options.py\", line 593, in _get_data_in_date_range\n    frame = getattr(self, nam)\nAttributeError: 'Options' object has no attribute 'calls160408'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas_datareader/yahoo/options.py\", line 216, in _get_option_data\n    frames = getattr(self, frame_name)\nAttributeError: 'Options' object has no attribute '_frames160408'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"<stdin>\", line 1, in <module>\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas_datareader/yahoo/options.py\", line 581, in get_all_data\n    return self._get_data_in_date_range(dates=expiry_dates, call=call, put=put)\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas_datareader/yahoo/options.py\", line 595, in _get_data_in_date_range\n    frame = self._get_option_data(expiry=expiry_date, name=name)\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas_datareader/yahoo/options.py\", line 218, in _get_option_data\n    frames = self._get_option_frames_from_yahoo(expiry)\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas_datareader/yahoo/options.py\", line 138, in _get_option_frames_from_yahoo\n    option_frames = self._option_frames_from_url(url)\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas_datareader/yahoo/options.py\", line 159, in _option_frames_from_url\n    frames = read_html(url)\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas/io/html.py\", line 866, in read_html\n    parse_dates, tupleize_cols, thousands, attrs, encoding)\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas/io/html.py\", line 718, in _parse\n    parser = _parser_dispatch(flav)\n  File \"/Users/hirofumi.yashima/anaconda/lib/python3.5/site-packages/pandas/io/html.py\", line 663, in _parser_dispatch\n    raise ImportError(\"html5lib not found, please install it\")\nImportError: html5lib not found, please install it\n>>>\n```\n\n* __get_all_data\u30e1\u30bd\u30c3\u30c9\u3092\u547c\u3073\u51fa\u3059\u3068\u3001html5lib\u30e2\u30b8\u30e5\u30fc\u30eb\u304c\u306a\u3044\u3068\u8a00\u308f\u308c\u308b\u3002__\n\n```{python:Python3}\n>>> import html5lib\nTraceback (most recent call last):\n  File \"<stdin>\", line 1, in <module>\nImportError: No module named 'html5lib'\n>>>\n>>> quit()\n```\n\n* __conda install \u3067 Anaconda \u306b \u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3059\u308b__\n\n```{bash:MacBook Terminal}\n~ $\n~ $ conda install html5lib\nFetching package metadata: ....\nSolving package specifications: .........\n\nPackage plan for installation in environment\n```\n\n* __\u4eca\u5ea6\u306f\u3001get_all_data\u30e1\u30bd\u30c3\u30c9 \u3092 \u5b9f\u884c\u3067\u304d\u305f__\n\n```{python:Python3}\n>>> import html5lib\n>>> from pandas_datareader.data import Options\n>>>\n>>> aapl = Options('aapl', 'yahoo')\n>>> data = aapl.get_all_data()\n>>> print(type(data))\n<class 'pandas.core.frame.DataFrame'>\n>>>\n>>> print(data.columns)\nIndex(['Last', 'Bid', 'Ask', 'Chg', 'PctChg', 'Vol', 'Open_Int', 'IV', 'Root',\n       'IsNonstandard', 'Underlying', 'Underlying_Price', 'Quote_Time'],\n      dtype='object')\n>>>\n>>> print(data.iloc[0:5, 0:5])\n                                            Last  Bid    Ask  Chg PctChg\nStrike Expiry     Type Symbol\n10.0   2016-06-17 call AAPL160617C00010000     0   91  91.45    0  0.00%\n                  put  AAPL160617P00010000     0    0   0.03    0  0.00%\n12.5   2016-06-17 put  AAPL160617P00012500     0    0   0.03    0  0.00%\n15.0   2016-06-17 call AAPL160617C00015000     0   86  86.45    0  0.00%\n                  put  AAPL160617P00015000     0    0   0.03    0  0.00%\n>>>\n```\n", "tags": ["Python", "python3", "\u6642\u7cfb\u5217\u89e3\u6790", "\u7d71\u8a08\u5b66", "statistics"]}